meta-mender-st-stm32mp/recipes-bsp/stm32mp-uboot-scr/files/boot.cmd.in

```diff
@@ -18,7 +18,7 @@ mmc dev ${mender_uboot_dev}
 run scan_m4fw
 
 load ${mender_uboot_root} ${kernel_addr_r} /boot/@@KERNEL_IMAGETYPE@@
-load ${mender_uboot_root} ${fdt_addr_r} /boot/stm32mp157d-dk1.dtb
+load ${mender_uboot_root} ${fdt_addr_r} /boot/@@MACHINE@@.dtb
```

Comment:
Yes, maybe you're right.  Although the "legacy support for specifying the full path to the device tree" would break things.

I actually have a bigger problem in that my bsp layer seems to contain a list for `KERNEL_DEVICETREE`, which doesn't seem to be "to spec" but does appear to mostly work everywhere else (and would not work with your suggested change).  The gpt image chokes on the list [here](https://github.com/mendersoftware/meta-mender-community/blob/kirkstone/meta-mender-st-stm32mp/classes/stm32mp-gptimg.bbclass#L12) and the `cp` command spits out a bunch of errors.  I can work around it by changing that command to treat the variable as a list (which would probably also be fine for the "normal" case of a single dtb):

```
for devicetree in ${KERNEL_DEVICETREE}; do
    cp ${DEPLOY_DIR_IMAGE}/kernel/${devicetree} ${WORKDIR}/bootfs/
done
```

And then I run into my [`fip` images](https://github.com/mendersoftware/meta-mender-community/blob/kirkstone/meta-mender-st-stm32mp/classes/stm32mp-gptimg.bbclass#L28) are "optee" not "trusted".

I was thinking that that this was a simple variable swap for literal string improvement, but it seems to be bigger than that.

In the end, I think I'm going to have to use a strategy to patch this class file in my local build environment, which is unfortunate but maybe the path of least resistance.

Feel free to close this MR or use whatever bits of it that you see fit.

Reply:
Generally speaking, have you looked at the octavo systems integration? It is essentially a STM32MP too, but using the generic OpenEmbedded setup strategy instead of STs setup magic. Maybe you can fit your needs in there more easily? In the long run, I would actually favor the OE way.

Reply:
I haven't looked at octavo, but on the surface I can't quite tell how their solution is different than any other OEM making a SOM-like thing.  I agree with you on using generic OE as much as possible though, and have spent quite a lot of time reproducing/replacing bits of vendor toolchains using open/upstream sources.